### PR TITLE
2.3.0-beta.3 release

### PR DESCRIPTION
**Pull request by azarovalex**
This PR contains a couple of changes:
- [x] Updated NN to 87.0.5
- [x] Removed unsupported "historyAutorecording" NN flag
- [x] Tests locale is hardcoded to "en" in the xcscheme
- [x] Fixed some compiler warnings in tests about unused variables
- [x] Off route test was changed to a more correct scenario
- [x] SDK version bump

### CODE REVIEW

File: Sources/MapboxCoreNavigation/NativeHandlersFactory.swift [hunk 1]
```diff
@@ -93,15 +93,7 @@ class NativeHandlersFactory {
     }()
     
     lazy var configHandle: ConfigHandle = {
-        let historyAutorecordingConfig = [
```
Comment (MaximAlien) [inline Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:96]: Does auto-recording work by default now?

Comment (azarovalex) [reply Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:96]: It's disabled by default, you need to call `startRecording` explicitly now.

File: Tests/MapboxNavigationTests/CarPlayManagerSpec.swift [hunk 2]
```diff
@@ -173,7 +173,7 @@ class CarPlayManagerSpec: QuickSpec {
                 }
             }
         }
-        
+#if arch(x86_64) && canImport(Darwin)
```
Comment (azarovalex) [inline Tests/MapboxNavigationTests/CarPlayManagerSpec.swift:176]: This test contains a `throwAssertion()` call which is not supported on M1 macs.
https://github.com/Quick/Nimble/blob/e4aa2014de13b90b06eaf183c67be3e390e927da/Sources/Nimble/Matchers/ThrowAssertion.swift#L46

File: Tests/MapboxCoreNavigationTests/NavigationServiceTests.swift [hunk 3]
```diff
@@ -105,14 +105,17 @@ class NavigationServiceTests: TestCase {
...
-            CLLocation(coordinate: $0.element,
-                       altitude: -1,
-                       horizontalAccuracy: 10,
-                       verticalAccuracy: -1,
-                       course: -1,
-                       speed: 10,
-                       timestamp: now + $0.offset)
+        let locationsOnRoute = coordinatesOnRoute.enumerated().map { tuple -> CLLocation in
+            // NavNative can filter locations if the disntance diff looks too big for the current speed and time diff
+            // so we use a big time delta here
```
Comment (MaximAlien) [inline Tests/MapboxCoreNavigationTests/NavigationServiceTests.swift:110]: Nit:
```suggestion
            // NavNative can filter locations if the distance diff looks too big for the current speed and time diff
            // so we use a big time delta here
```

File: MapboxNavigation-SPM.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 4]
```diff
@@ -69,8 +69,8 @@
         "repositoryURL": "https://github.com/mapbox/mapbox-navigation-native-ios.git",
         "state": {
           "branch": null,
-          "revision": "86c79b6b8fd5b4c8a95300e2f0a5e214c24c75f0",
```
Comment (MaximAlien) [inline MapboxNavigation-SPM.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:72]: I guess it'd make sense to update `Cartfile.resolved` as well.

**Review by MaximAlien — APPROVED** [review]

**Review by ShanMa1991 — APPROVED** [review]

**Pull request merged**